Addons/testExoticQuadrature.hpp

```diff
@@ -0,0 +1,223 @@
+            passed = false;
+        }
+        if (ct.getNumLevels() != n) {
+            std::cout << "ERROR: Test failed in testCtAttributes() for shift = " << shift << " on getNumLevels()" << std::endl;
+            passed = false;
+        }
+        for (int i=0; i<n; i++) {
+            if (ct.getQExact(i) != 2*(i+1)-1) {
+                std::cout << "ERROR: Test failed in testCtAttributes() for shift = " << shift << ", i = " << i
+                          << " on getQExact()" << std::endl;
+                passed = false;
+            }
+        }
+    }
+    return passed;
+}
+
+// Integrates the function f(x) * sinc(freq * x) over [-1, 1]  using Exotic
+// quadrature at level n and a specified shift. Then, compares this integral
+// against a given exact integral value.
+inline bool wrapSincTest1D(std::function<double(double)> f, int level, double freq, double shift, double exact_integral) {
+    bool passed = true;
+    double precision = 1e-10;
+    auto sinc = [freq](double x)->double {return (x == 0.0 ? 1.0 : sin(freq * x) / (freq * x));};
+    TasGrid::CustomTabulated ct = TasGrid::getExoticQuadrature(level, shift, sinc);
+    TasGrid::TasmanianSparseGrid sg;
+    sg.makeGlobalGrid(1, 1, 2*level-1, TasGrid::type_qptotal, std::move(ct));
+    std::vector<double> quad_points = sg.getPoints();
+    std::vector<double> quad_weights = sg.getQuadratureWeights();
+    double approx_integral = 0.0;
+    assert(quad_weights.size() == quad_points.size());
+    for (size_t i=0; i<quad_weights.size(); i++) {
+        approx_integral += f(quad_points[i]) * quad_weights[i];
+    }
+    if (std::abs(approx_integral - exact_integral) > precision) {
+        std::cout << "ERROR: " << std::setprecision(16) << "Computed integral value " << approx_integral
+                  << " does not match exact integral " << exact_integral << std::endl;
+        passed = false;
+    }
+    return passed;
+}
+
+// Test the accuracy of the exotic quadrature over some 1D functions.
+inline bool testExpMx2_sinc1_shift0() {
+    auto f = [](double x)->double {return std::exp(-x*x);};
+    bool passed = wrapSincTest1D(f, 20, 1.0, 0.0, 1.4321357541271255);
+    if (not passed) {
+        std::cout << "ERROR: Test failed in testExpMx2_sinc1_shift0()" << std::endl;
+    }
+    return passed;
+}
+inline bool testExpMx2_sinc1_shift1() {
+    auto f = [](double x)->double {return std::exp(-x*x);};
+    bool passed = wrapSincTest1D(f, 20, 1.0, 1.0, 1.4321357541271255);
+    if (not passed) {
+        std::cout << "ERROR: Test failed in testExpMx2_sinc1_shift1()" << std::endl;
+    }
+    return passed;
+}
+inline bool testExpMx2_sinc10_shift1() {
+    auto f = [](double x)->double {return std::exp(-x*x);};
+    bool passed = wrapSincTest1D(f, 20, 10.0, 1.0, 0.32099682841103033);
+    if (not passed) {
+        std::cout << "ERROR: Test failed in testExpMx2_sinc10_shift1()" << std::endl;
+    }
+    return passed;
+}
+inline bool testExpMx2_sinc100_shift1() {
```

Comment:
This is looking much better now.

Right here, you don't need all of those functions, all that those do is define a test function `f`, call `wrapSincTest1D()` and then print an error message of the rest fails. If you want to see the specific method being called, just add a string parameter to `wrapSincTest1D()` and do the cout statement there. Then call the wrapper directly by inlining the lambda expressions:
```
pass = pass and wrapSincTest1D([](double x)->double { return std::exp(-x*x); }, 20, 100.0, 1.0, 0.031353648322695503);
```